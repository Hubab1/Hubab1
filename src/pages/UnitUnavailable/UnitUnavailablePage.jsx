import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'constants/constants';
import { prettyFormatPhoneNumber } from 'utils/misc';
import captureRoute from 'utils/captureRoute';

import UnauthenticatedPage from 'common-components/Pages/UnauthenticatedPage/UnauthenticatedPage';
import { H1, SpacedH3, P } from 'assets/styles';
import tooslow from 'assets/images/tooslow.png';

export const Img = styled.img`
    padding-top: 10px;
    margin-bottom: 76px;
`;

export const UnitUnavailablePage = ({ configuration, profile }) => {
    if (!configuration) {
        return null;
    }
    const unit = configuration?.unit || profile?.unit;
    if (!unit) {
        return null;
    }
    const Wrapper = profile ? React.Fragment : UnauthenticatedPage;

    return (
        <Wrapper>
            <H1>Someone Beat You To It!</H1>
            <SpacedH3>Sorry, unit {unit.unit_number} is no longer available.</SpacedH3>
            <Img src={tooslow} />
            <div className="align-left">
                <P>{`Looks like someone else swiped this unit off the market. Don't worry though, we can help you find a similar one!`}</P>
                <br />
                <P bold>
                    Call us at{' '}
                    <a href={`tel:${configuration.community.contact_phone}`}>
                        {prettyFormatPhoneNumber(configuration.community.contact_phone)}
                    </a>
                    &nbsp;so we can discuss other options.
                </P>
            </div>
        </Wrapper>
    );
};

UnitUnavailablePage.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
};

const mapStateToProps = (state) => ({
    profile: state.renterProfile,
    configuration: state.configuration,
});

export default connect(mapStateToProps)(captureRoute(UnitUnavailablePage, ROUTES.UNIT_UNAVAILABLE));
