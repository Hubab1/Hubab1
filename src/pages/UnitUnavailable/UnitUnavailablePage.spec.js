import React from 'react';
import { shallow } from 'enzyme';
import { UnitUnavailablePage } from './UnitUnavailablePage';

describe('UnitUnavailablePage', () => {
    let defaultProps;

    beforeEach(() => {
        defaultProps = {
            profile: {},
            configuration: {
                community: {
                    contact_phone: '123456',
                },
            },
        };
    });

    describe('missing profile', () => {
        it('returns null', () => {
            const props = {
                ...defaultProps,
                profile: null,
            };
            const wrapper = shallow(<UnitUnavailablePage {...props} />);
            expect(wrapper.instance()).toBe(null);
        });
    });

    describe('missing configuration', () => {
        it('returns null', () => {
            const props = {
                ...defaultProps,
                configuration: null,
            };
            const wrapper = shallow(<UnitUnavailablePage {...props} />);
            expect(wrapper.instance()).toBe(null);
        });
    });

    describe('signed in', () => {
        it('matches snapshot', () => {
            const props = {
                ...defaultProps,
                profile: {
                    unit: {
                        unit_number: 123,
                    },
                },
            };
            const wrapper = shallow(<UnitUnavailablePage {...props} />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });

    describe('signed out', () => {
        it('matches snapshot', () => {
            const props = {
                ...defaultProps,
                configuration: {
                    ...defaultProps.configuration,
                    unit: {
                        unit_number: 123,
                    },
                },
            };
            const wrapper = shallow(<UnitUnavailablePage {...props} />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
