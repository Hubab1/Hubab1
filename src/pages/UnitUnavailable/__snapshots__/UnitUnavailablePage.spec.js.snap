// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitUnavailablePage signed in matches snapshot 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Someone Beat You To It!
  </ForwardRef(render)>
  <ForwardRef(render)>
    Sorry, unit 
    123
     is no longer available.
  </ForwardRef(render)>
  <ForwardRef(render)
    src="tooslow.png"
  />
  <div
    className="align-left"
  >
    <ForwardRef(render)>
      Looks like someone else swiped this unit off the market. Don't worry though, we can help you find a similar one!
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)
      bold={true}
    >
      Call us at
       
      <a
        href="tel:123456"
      >
        123456
      </a>
       so we can discuss other options.
    </ForwardRef(render)>
  </div>
</React.Fragment>
`;

exports[`UnitUnavailablePage signed out matches snapshot 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Someone Beat You To It!
  </ForwardRef(render)>
  <ForwardRef(render)>
    Sorry, unit 
    123
     is no longer available.
  </ForwardRef(render)>
  <ForwardRef(render)
    src="tooslow.png"
  />
  <div
    className="align-left"
  >
    <ForwardRef(render)>
      Looks like someone else swiped this unit off the market. Don't worry though, we can help you find a similar one!
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)
      bold={true}
    >
      Call us at
       
      <a
        href="tel:123456"
      >
        123456
      </a>
       so we can discuss other options.
    </ForwardRef(render)>
  </div>
</React.Fragment>
`;
